sdk/table/azure-table/azure/table/_table_service_client.py

```diff
@@ -196,6 +203,12 @@ def delete_table(
         :return: None
         :rtype: None
         """
+        if re.match("^[a-zA-Z]{1}[a-zA-Z0-9]{2,62}$", table_name) is None:
```

Comment:
wonder if we can refactor this code into a function like `is_valid_table_name(table_name)` to reduce duplicate code across clients